app/src/display/widgets/label.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/display.h>
+#include <zmk/display/widgets/label.h>
+#include <zmk/events/activity_state_changed.h>
+#include <zmk/event_manager.h>
+
+#define WIDGET_LABEL_TEXT_MAX 7
+
+BUILD_ASSERT(sizeof(CONFIG_ZMK_WIDGET_LABEL_TEXT) - 1 <= WIDGET_LABEL_TEXT_MAX,
+             "ERROR: Widget label text length is too long. Max length: 7");
+
+static sys_slist_t widgets = SYS_SLIST_STATIC_INIT(&widgets);
+
+struct label_state {};
+
+static struct label_state get_state(const zmk_event_t *_eh) { return (struct label_state){}; }
+
+static void set_label_symbol(lv_obj_t *label) {
+    lv_label_set_text(label, CONFIG_ZMK_WIDGET_LABEL_TEXT);
+}
+
+static void label_update_cb() {
+    struct zmk_widget_label *widget;
+    SYS_SLIST_FOR_EACH_CONTAINER(&widgets, widget, node) { set_label_symbol(widget->obj); }
+}
+
+ZMK_DISPLAY_WIDGET_LISTENER(widget_label, struct label_state, label_update_cb, get_state)
+ZMK_SUBSCRIPTION(widget_label, zmk_activity_state_changed);
+
```

Comment:
Does this really need to subscribe to anything and update when activity state changes? Seems like this can be initialized once in the init with the configured text, and be done.

Reply:
Hey @petejohanson 
You are absolutely right no subscription is needed.
As you might understood this was my testing widget to learn about how subs/listeners work inside zmk, hence their existance there, as I tried to mimick functionality like rest widgets.
Removing them, along with other not needed parts.